Bottom: c08abed84c43af42d9c8515622549457c6bd3c56
Top:    460c31bdd7ebd918773d2209efd95bf230bc7bb4
Author: Jared Casper <jaredcasper@gmail.com>
Date:   2009-10-21 19:55:36 +0100

Fix a bug in gtk hid which caused layer buttons to get confused (sourceforge bug 1988951)

In ghid_layer_enable_buttons_update, the layer_buttons array was being
indexed using a counter which stoped at the current max_layer, which
was wrong.  Changed this to use the constants LAYER_BUTTON_*.

Also made the handling of the silk and rats buttons match the
(slightly more correct) handling of the other buttons in
layer_enable_button_cb.


---

diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index ca17446..b950a25 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -1380,14 +1380,14 @@ layer_enable_button_cb (GtkWidget * widget, gpointer data)
   switch (layer)
     {
     case LAYER_BUTTON_SILK:
-      PCB->ElementOn = !PCB->ElementOn;
+      PCB->ElementOn = active;
       PCB->Data->SILKLAYER.On = PCB->ElementOn;
       PCB->Data->BACKSILKLAYER.On = PCB->ElementOn;
       redraw = 1;
       break;
 
     case LAYER_BUTTON_RATS:
-      PCB->RatOn = !PCB->RatOn;
+      PCB->RatOn = active;
       redraw = 1;
       break;
 
@@ -1649,23 +1649,23 @@ ghid_layer_enable_buttons_update (void)
   /* Buttons for elements (silk), rats, pins, vias, and far side don't
      |  change labels.
    */
-  lb = &layer_buttons[i++];	/* LAYER_BUTTON_SILK */
+  lb = &layer_buttons[LAYER_BUTTON_SILK];
   gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (lb->layer_enable_button),
 				PCB->ElementOn);
 
-  lb = &layer_buttons[i++];	/* LAYER_BUTTON_RATS */
+  lb = &layer_buttons[LAYER_BUTTON_RATS];
   gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (lb->layer_enable_button),
 				PCB->RatOn);
 
-  lb = &layer_buttons[i++];	/* pins/pads    */
+  lb = &layer_buttons[LAYER_BUTTON_PINS];
   gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (lb->layer_enable_button),
 				PCB->PinOn);
 
-  lb = &layer_buttons[i++];	/* vias         */
+  lb = &layer_buttons[LAYER_BUTTON_VIAS];
   gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (lb->layer_enable_button),
 				PCB->ViaOn);
 
-  lb = &layer_buttons[i++];	/* far side     */
+  lb = &layer_buttons[LAYER_BUTTON_FARSIDE];
   gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (lb->layer_enable_button),
 				PCB->InvisibleObjectsOn);
   layer_enable_button_cb_hold_off = FALSE;
